common/internal_model/src/main/scala/weco/catalogue/internal_model/work/Work.scala

```diff
@@ -107,9 +108,23 @@ case class WorkData[State <: DataState](
   items: List[Item[State#MaybeId]] = Nil,
   holdings: List[Holdings] = Nil,
   collectionPath: Option[CollectionPath] = None,
+  referenceNumber: Option[ReferenceNumber] = None,
   imageData: List[ImageData[State#Id]] = Nil,
   workType: WorkType = WorkType.Standard,
-)
+) {
+
+  // We have separate fields for collectionPath and referenceNumber so we
+  // can set a referenceNumber which isn't part of a collection.
+  // (e.g. an Iconographic number on Sierra records.)
+  //
+  // If a collectionPath is set, we should always require a referenceNumber
+  // to be set, based on its value.
```

Comment:
Why is this a requirement?

Reply:
All collection paths are reference numbers (e.g. `PP/ALM/1`), but not all reference numbers are collection paths (e.g. `12345i`). This assertion was an attempt to encode that – if we've got a collection path, we should always set a reference number – but it breaks the Scalacheck tests and is not that important, so I've taken it out.

Reply:
Could you define what a reference number is?

Reply:
There's a longer thread in Slack here: https://wellcome.slack.com/archives/C294K7D5M/p1620733997159000

> when they are the reference we have written on the actual item, they should be the referenceNumber
>
> Ie there is a number written in pencil on the old piece of paper and it is XXXXX - that’s the reference number, as it’s what people will see when they have the object in their hands

Reply:
Could have that on the class as it's more accessible to future devs.